HOTT-1079 Fixes on switch banner

alexdesi opened the pull request:
### Jira link
https://transformuk.atlassian.net/browse/HOTT-1079


### What?
- This fixes the double switch banner issue and hides the banner on the Meursing code finder page (and other pages).
- Also, this makes a small refactor on the shared/search partial

### Why?
Because we hate BUGS!!! :rage1:

alexdesi committed:
HOTT-1079 This fixes the double switch banner issue and
hides the banner on the Meursing code finder page.

jebw commented on app/controllers/application_controller.rb:
```diff
@@ -73,8 +73,8 @@ def set_last_updated
 
-  def set_enable_service_switch_banner_in_action
-    @enable_service_switch_banner_in_action = true
+  def set_switch_banner
```
I know I'm going to seem really picky here - but `switch_banner` doesn't make any reference to what its switching - a potential point of confusion for future developers.

Can this be renamed to something like `service_switch_banner` which is a bit more specific but still short?

alexdesi replied:
👍

jebw reviewed (commented):
Only comment is around naming?

alexdesi committed:
HOTT-1079 This fixes the following:
- double Switch banner in some pages
- no Switch banner in other pages
- refactor: it extract the breadcrumb and switch banner from the search partial

alexdesi commented on app/views/layouts/application.html.erb:
```diff
@@ -76,11 +76,13 @@
       <%= render 'shared/phase_banner' if ENV['NOTICE_BANNER'] %>
       <%= yield :before_main_content %>
 
+      <%= render 'shared/top_breadcrumbs' %>
+      <%= render 'shared/switch_banner' %>
```
Those partials have been extracted from the `shared/search` partial.
(why did the search contain the breadcrumb and switch? 🤷 )

willfish replied:
So much better

willfish commented on app/controllers/application_controller.rb:
```diff
@@ -73,8 +73,16 @@ def set_last_updated
 
-  def set_enable_service_switch_banner_in_action
-    @enable_service_switch_banner_in_action = true
+  def set_switch_service_banner
```
```suggestion
  def enable_switch_service_banner
```

willfish commented on app/controllers/application_controller.rb:
```diff
@@ -7,7 +7,7 @@ class ApplicationController < ActionController::Base
 
   before_action :set_cache
-  before_action :set_enable_service_switch_banner_in_action
+  before_action :set_switch_service_banner
```
```suggestion
  before_action :enable_switch_service_banner
```

jebw replied:
I'll add my +1 to will's suggestion

willfish commented on app/controllers/pages_controller.rb:
```diff
@@ -1,6 +1,7 @@
 class PagesController < ApplicationController
   before_action do
     @tariff_last_updated = nil
+
```
Do you want this extra line :)?

willfish commented on app/controllers/pages_controller.rb:
```diff
@@ -15,8 +15,8 @@ def opensearch
   end
 
   def tools
+    set_switch_service_banner
```
```suggestion
    enable_switch_service_banner
```

willfish replied:
Actually isn't this the default so I don't think you need this?

willfish commented on app/controllers/search_controller.rb:
```diff
@@ -46,8 +46,6 @@ def quota_search
 
     form = QuotaSearchForm.new(params)
     @result = QuotaSearchPresenter.new(form)
-    @enable_service_switch_banner_in_action = false
```
Shouldn't you replace this with a call to `disable_switch_service_banner`

alexdesi replied:
Good catch! 
Actually here the banner was enabled here, so I removed that line.

(it has been enabled in various places too)

alexdesi committed:
HOTT-1079 Fixes CSS for the Switch Banner (bottom) for mobile version.
